torch/csrc/distributed/c10d/symm_mem/NVSHMEMSymmetricMemory.cu

```diff
@@ -345,6 +346,7 @@ class NVSHMEMSymmetricMemoryAllocator : public SymmetricMemoryAllocator {
 
     initialize_nvshmem_with_store(store, rank, world_size);
     auto ptr = nvshmem_malloc(size);
+    TORCH_CHECK(ptr != nullptr, "nvshmem_malloc failed");
```

Comment:
if `size` is 0 (which is legal allocation request) we shouldn't error.

Reply:
Good point, thanks!